nix/scripts/setup.sh

```diff
@@ -1,51 +1,68 @@
 #!/usr/bin/env bash
+# This script installs a specific version of Nix.
+set -eo pipefail
 
 GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
 source "${GIT_ROOT}/scripts/colors.sh"
 
-NIX_VERSION="2.3.12"
+NIX_VERSION="2.5.1"
 NIX_INSTALL_URL="https://nixos.org/releases/nix/nix-${NIX_VERSION}/install"
-NIX_INSTALL_SHA256="468a49a1cef293d59508bb3b62625dfcd99ec00334a14309f125cf8de513d5f1"
+NIX_INSTALL_SHA256="e265dfd8e80223633a9726009b42c534ac3d5f2b6da5ad6432ca1f6ea88206d0"
 NIX_INSTALL_PATH="/tmp/nix-install-${NIX_VERSION}"
 
-function install_nix() {
-  # Don't break people's profiles
-  export NIX_INSTALLER_NO_MODIFY_PROFILE=1
-  # Fix for installing on MacOS Catalina
-  export NIX_IGNORE_SYMLINK_STORE=1
-  # Download installer and verify SHA256
-  curl -s "${NIX_INSTALL_URL}" -o "${NIX_INSTALL_PATH}"
-  echo "${NIX_INSTALL_SHA256}  ${NIX_INSTALL_PATH}" | sha256sum -c
-  chmod +x "${NIX_INSTALL_PATH}"
-  # Run the installer
-  "${NIX_INSTALL_PATH}" --no-daemon
-  if [ $? -eq 0 ]; then
-    echo -e "${GRN}The Nix package manager was successfully installed.${RST}"
-  else
-    echo -e "${RED}Failed to install Nix package manager!${RST}" >&2
-    echo "Please see: https://nixos.org/nix/manual/#chap-installation" >&2
-    exit 1
-  fi
+install_nix() {
+    # Don't break people's profiles
+    export NIX_INSTALLER_NO_MODIFY_PROFILE=1
+
+    # Download installer and verify SHA256
+    curl -s "${NIX_INSTALL_URL}" -o "${NIX_INSTALL_PATH}"
+    echo "${NIX_INSTALL_SHA256}  ${NIX_INSTALL_PATH}" | sha256sum -c
+    chmod +x "${NIX_INSTALL_PATH}"
+
+    # Identify installation type
+    if [[ -z "${NIX_INSTALL_OPTS}" ]]; then
+        if [[ "${UNAME}" == "Linux" ]]; then
```

Comment:
when running the script with no parameters directly as `nix/scripts/setup.sh`, the installation options are listed. `UNAME` seems to be undefined (I don't see it defined anywhere in this script)

Reply:
Ah, you're right, I must have lost `UNAME=$(uname -s)` during some cleanup.